server/services/caseloadService.ts

```diff
@@ -0,0 +1,36 @@
+import CommunityService from './communityService'
+import PrisonerService from './prisonerService'
+import { CommunityApiManagedOffender } from '../@types/communityClientTypes'
+import { ManagedCase } from '../@types/managedCase'
+
+export default class CaseloadService {
+  constructor(private readonly prisonerService: PrisonerService, private readonly communityService: CommunityService) {}
+
+  async getStaffCaseload(username: string): Promise<ManagedCase[]> {
+    // TODO: Change this - How should we handle the error for when users do not have an account in delius?
+    let managedOffenders: CommunityApiManagedOffender[]
+    try {
+      const { staffIdentifier } = await this.communityService.getStaffDetail(username)
+      managedOffenders = await this.communityService.getManagedOffenders(staffIdentifier)
+    } catch (e) {
+      if (e.status === 404) {
+        managedOffenders = []
+      }
+    }
+
+    const caseloadNomisIds = managedOffenders
+      .filter(offender => offender.currentOm)
+      .filter(offender => offender.nomsNumber)
+      .map(offender => offender.nomsNumber)
+
+    return (await this.prisonerService.searchPrisonersByNomisIds(username, caseloadNomisIds)).map(offender => {
+      const matchingDeliusCase = managedOffenders.find(deliusCase => deliusCase.nomsNumber === offender.prisonerNumber)
+      return {
+        ...matchingDeliusCase,
+        ...offender,
+      } as ManagedCase
+
+      // TODO Cache this by username?
+    })
```

Comment:
Can cache in redis later and wrap the get with a redis lookup first - good examples in other services to reuse.